Compile error WITH_SERIALPORT=false

### What is the bug or the crash?

Compile error `WITH_SERIALPORT=false` 
```
sip-build: /var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1_build/python/core/core.sip: line 101: column 9: 'QtSerialPort/QtSerialPortmod.sip' could not be found
ninja: build stopped: subcommand failed.
```

### Steps to reproduce the issue

compile 3.36.1 using option `WITH_SERIALPORT=false`
- configure 
  ```
  cmake -C /var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DQGIS_MANUAL_SUBDIR=share/man/ -DQGIS_LIB_SUBDIR=lib64 -DQGIS_PLUGIN_SUBDIR=lib64/qgis -DQWT_INCLUDE_DIR=/usr/include/qwt6 -DQWT_LIBRARY=/usr/lib64/libqwt6-qt5.so -DQGIS_QML_SUBDIR=/usr/lib64/qt5/qml -DPEDANTIC=OFF -DUSE_CCACHE=OFF -DWITH_ANALYSIS=ON -DWITH_APIDOC=OFF -DWITH_GUI=ON -DWITH_INTERNAL_MDAL=ON -DWITH_QSPATIALITE=yes -DENABLE_TESTS=yes -DWITH_3D=yes -DWITH_GSL=yes -DCMAKE_DISABLE_FIND_PACKAGE_HDF5=OFF -DWITH_SERVER=yes -DCMAKE_DISABLE_FIND_PACKAGE_NetCDF=OFF -DUSE_OPENCL=yes -DWITH_ORACLE=no -DWITH_QWTPOLAR=no -DWITH_PDAL=yes -DWITH_COPC=yes -DWITH_POSTGRESQL=yes -DWITH_BINDINGS=yes -DWITH_CUSTOM_WIDGETS=yes -DWITH_QUICK=yes -DWITH_QTSERIALPORT=no -DWITH_QTWEBKIT=OFF -DBUILD_WITH_QT6=no -DGRASS_PREFIX8=/usr/lib64/grass83 -DWITH_GRASS8=ON -DBINDINGS_GLOBAL_INSTALL=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1_build/gentoo_toolchain.cmake /var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1
  ```
- compile 
  ```
  ninja -v j33 -l0
  ```



### Versions

- 3.36.1
- or basically any version since https://github.com/qgis/QGIS/commit/9195dbd05750498f08a473bdd4ac23e660743531 fails with this very error



### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

cc @jtornero this issue seems to be related to 9195dbd05750498f08a473bdd4ac23e660743531 which assumes unconditionally `WITH_SERIALPORT=true` - which isn't the case. AFAICT QtSerialport should be compiled conditionally only upon `%If (HAVE_QTSERIALPORT)` as re-fixed by https://github.com/qgis/QGIS/commit/ecdb1e2989e3bf86978590fc4fd724b18b8dde27. 
this concerns at least https://github.com/qgis/QGIS/commit/9195dbd05750498f08a473bdd4ac23e660743531#diff-9ad07175b4e5bd06fbdf8f4a8f69b63d4c674a3bbcca791650067f051c761be5R101 and https://github.com/qgis/QGIS/commit/9195dbd05750498f08a473bdd4ac23e660743531#diff-a71aa025b44462573371fd6c18befdff13db223ff9736f1dfea29fd78774902eR101 and probably also the python code

... or would you vote to drop support for this option and make qtserial a requirement?

--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -116,7 +116,7 @@ jobs:
           echo QT_VERSION: ${QT_VERSION}
 
       - name: Login to Docker Hub
-        if: ${{ github.event_name == 'push' }}
+        if: ${{ github.event_name == 'push' && github.actor == 'qgis' }}
         uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
@@ -129,7 +129,7 @@ jobs:
           context: .
           file: .docker/qgis3-qt${{ matrix.qt-version }}-build-deps.dockerfile
           tags: qgis/qgis3-build-deps-${{ matrix.distro-version }}-qt${{ matrix.qt-version }}:${{ github.event.pull_request.base.ref || github.ref_name }}
-          push: ${{ github.event_name == 'push' }}
+          push: ${{ github.event_name == 'push' && github.actor == 'qgis' }}
           pull: true
           build-args:
             DISTRO_VERSION=${{ matrix.distro-version }}
@@ -347,7 +347,7 @@ jobs:
           echo CTEST_BUILD_NAME: ${CTEST_BUILD_NAME}
 
       - name: Login to Docker Hub
-        if: ${{ github.event_name == 'push' }}
+        if: ${{ github.event_name == 'push' && github.actor == 'qgis' }}
         uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
@@ -360,7 +360,7 @@ jobs:
           context: .
           file: .docker/qgis3-qt${{ matrix.qt-version }}-build-deps.dockerfile
           tags: qgis/qgis3-qt${{ matrix.qt-version }}-build-deps-bin-only:${{ github.event.pull_request.base.ref || github.ref_name }}
-          push: ${{ github.event_name == 'push' }}
+          push: ${{ github.event_name == 'push' && github.actor == 'qgis' }}
           pull: true
           target: ${{ matrix.docker-target }}
           build-args:
@@ -458,7 +458,7 @@ jobs:
           fetch-depth: 2
 
       - name: Login to Docker Hub
-        if: ${{ github.event_name == 'push' }}
+        if: ${{ github.event_name == 'push' && github.actor == 'qgis' }}
         uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
@@ -471,7 +471,7 @@ jobs:
           context: .
           file: .docker/qgis3-qt${{ matrix.qt-version }}-build-deps.dockerfile
           tags: qgis/qgis3-qt${{ matrix.qt-version }}-build-deps-bin-only:${{ github.event.pull_request.base.ref || github.ref_name }}
-          push: ${{ github.event_name == 'push' }}
+          push: ${{ github.event_name == 'push' && github.actor == 'qgis' }}
           pull: true
           target: ${{ matrix.docker-target }}
           build-args:

--- a/python/PyQt6/core/core.sip.in
+++ b/python/PyQt6/core/core.sip.in
@@ -91,15 +91,6 @@ done:
 
 %End
 
-%Import QtXml/QtXmlmod.sip
-%Import QtNetwork/QtNetworkmod.sip
-%Import QtSql/QtSqlmod.sip
-
-%Import QtPrintSupport/QtPrintSupportmod.sip
-%Import QtWidgets/QtWidgetsmod.sip
-%Import QtPositioning/QtPositioningmod.sip
-%Import QtSerialPort/QtSerialPortmod.sip
-
 %Feature HAVE_GUI
 %Feature HAVE_QTSERIALPORT
 %Feature HAVE_QTPRINTER
@@ -107,6 +98,18 @@ done:
 %Feature VECTOR_MAPPED_TYPE
 %Feature HAVE_WEBENGINE_SIP
 %Feature PYQT6
+
+%Import QtXml/QtXmlmod.sip
+%Import QtNetwork/QtNetworkmod.sip
+%Import QtSql/QtSqlmod.sip
+
+%Import QtPrintSupport/QtPrintSupportmod.sip
+%Import QtWidgets/QtWidgetsmod.sip
+%Import QtPositioning/QtPositioningmod.sip
+
+%If (HAVE_QTSERIALPORT)
+%Import QtSerialPort/QtSerialPortmod.sip
+%End
 
 %Include conversions.sip
 %Include qgsexception.sip

--- a/python/core/core.sip.in
+++ b/python/core/core.sip.in
@@ -91,6 +91,13 @@ done:
 
 %End
 
+%Feature HAVE_GUI
+%Feature HAVE_QTSERIALPORT
+%Feature HAVE_QTPRINTER
+%Feature ANDROID
+%Feature VECTOR_MAPPED_TYPE
+%Feature HAVE_WEBENGINE_SIP
+
 %Import QtXml/QtXmlmod.sip
 %Import QtNetwork/QtNetworkmod.sip
 %Import QtSql/QtSqlmod.sip
@@ -98,14 +105,10 @@ done:
 %Import QtPrintSupport/QtPrintSupportmod.sip
 %Import QtWidgets/QtWidgetsmod.sip
 %Import QtPositioning/QtPositioningmod.sip
-%Import QtSerialPort/QtSerialPortmod.sip
 
-%Feature HAVE_GUI
-%Feature HAVE_QTSERIALPORT
-%Feature HAVE_QTPRINTER
-%Feature ANDROID
-%Feature VECTOR_MAPPED_TYPE
-%Feature HAVE_WEBENGINE_SIP
+%If (HAVE_QTSERIALPORT)
+%Import QtSerialPort/QtSerialPortmod.sip
+%End
 
 %Include conversions.sip
 %Include qgsexception.sip

